src/core_modules/capture-core/components/common/bulkDataEntry/hooks/useBulkDataEntryConfigurations.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import { useCallback } from 'react';
+import { useQueryClient } from 'react-query';
+import { ReactQueryAppNamespace } from 'capture-core/utils/reactQueryHelpers';
+import { setBulkDataEntry, removeBulkDataEntry } from './bulkDataEntryStorageUtils';
+import { useBulkDataEntryFromIndexedDB } from './useBulkDataEntryFromIndexedDB';
+import { useBulkDataEntryDatastoreConfigurations } from './useBulkDataEntryDatastoreConfigurations';
+
+const getActiveBulkDataEntryConfiguration = (bulkDataEntryConfigurations, cachedBulkDataEntry) =>
+    bulkDataEntryConfigurations?.find(config => config.configKey === cachedBulkDataEntry?.activeList?.configKey);
```

Comment:
Let's make sure that we match the `programId` as well, ref: https://github.com/dhis2/capture-app/pull/4066#discussion_r2121161557

Reply:
The `bulkDataEntryConfigurations` returned by [useBulkDataEntryDatastoreConfigurations](https://github.com/dhis2/capture-app/pull/4000/files#diff-594bacedcc90bc99554cc0026c0c0a69723956a7ff5b7907166cf04b48059248R84) is already filtered and matched by the `programId`

Reply:
Ah, perfect, makes sense.